kratos/sources/kratos_parameters.cpp

```diff
@@ -1080,6 +1093,57 @@ void Parameters::ValidateDefaults(const Parameters& rDefaultParameters) const
 /***********************************************************************************/
 /***********************************************************************************/
 
+void Parameters::ValidateDefaultsSkipSubParameters(const Parameters& rDefaultParameters) const
+{
+    KRATOS_TRY
+
+    // Verifies that all the entries in the current parameters have a correspondance in the rDefaultParameters.
+    // If it is not the case throw an error
+    for (auto itr = this->mpValue->begin(); itr != this->mpValue->end(); ++itr) {
+        const std::string& r_item_name = itr.key();
+        if(itr->is_object()) {
```

Comment:
This whole function is just a copy paste of `ValidateDefaults` except this if condition. Can't this be done by passing a flag to this function if we want to do it recursively or not ? On python level we can expose two functions if that is really necessary.

Reply:
I don't know how to do this without "hardcoding", maybe splitting the function in smaller private ones...